code/datums/antagonists/datum_vampire.dm

```diff
@@ -0,0 +1,280 @@
+/datum/antagonist/vampire
+	var/usable_blood = 0
+	var/total_blood = 0
+	var/fullpower = FALSE
+	var/draining
+	var/list/objectives_given = list()
+
+	var/iscloaking = FALSE
+
+	var/list/powers = list() // list of current powers
+
+	var/obj/item/clothing/suit/draculacoat/coat
+
+	var/list/upgrade_tiers = list(
+		/obj/effect/proc_holder/spell/self/rejuvenate = 0,
+		/obj/effect/proc_holder/spell/targeted/hypnotise = 0,
+		/datum/vampire_passive/vision = 175,
+		/obj/effect/proc_holder/spell/self/shapeshift = 175,
+		/obj/effect/proc_holder/spell/self/cloak = 225,
+		/obj/effect/proc_holder/spell/targeted/disease = 275,
+		/obj/effect/proc_holder/spell/bats = 350,
+		/obj/effect/proc_holder/spell/self/batform = 350,
+		/obj/effect/proc_holder/spell/self/screech = 315,
+		/datum/vampire_passive/regen = 425,
+		/obj/effect/proc_holder/spell/targeted/ethereal_jaunt/mistform = 500,
+		/datum/vampire_passive/full = 666,
+		/obj/effect/proc_holder/spell/self/summon_coat = 666,
+		/obj/effect/proc_holder/spell/targeted/vampirize = 700,
+		/obj/effect/proc_holder/spell/self/revive = 800)
+
+/datum/antagonist/vampire/on_gain()
+	SSticker.mode.vampires += owner
+	give_objectives()
+	check_vampire_upgrade()
+	owner.special_role = "vampire"
+	owner.current.faction += "vampire"
+	SSticker.mode.update_vampire_icons_added(owner)
+	..()
+
+/datum/antagonist/vampire/on_removal()
+	remove_vampire_powers()
+	owner.current.faction -= "vampire"
+	SSticker.mode.vampires -= owner
+	owner.special_role = null
+	if(ishuman(owner.current))
+		var/mob/living/carbon/human/H = owner.current
+		if(owner && H.hud_used && H.hud_used.vamp_blood_display)
+			H.hud_used.vamp_blood_display.invisibility = INVISIBILITY_ABSTRACT
+	SSticker.mode.update_vampire_icons_removed(owner)
+	for(var/O in objectives_given)
+		owner.objectives -= O
+	LAZYCLEARLIST(objectives_given)
+	if(owner.current)
+		to_chat(owner.current,"<span class='userdanger'>Your powers have been quenched! You are no longer a vampire</span>")
+	owner.special_role = null
+	..()
+
+/datum/antagonist/vampire/greet()
+	to_chat(owner, "<span class='userdanger'>You are a Vampire!</span>")
+	to_chat(owner, "<span class='danger bold'>You are a creature of the night -- holy water, the chapel, and space will cause you to burn.</span>")

```

Comment:
just say "will burn you"

Reply:
nitpicking 😕

Reply:
Actually the mechanic literally sets you on fire. "Will burn you" is vague and misleading.  "will set you aflame" would probably be the  best.